packages/sdk/src/protocols/base_protocol.ts

```diff
@@ -232,6 +258,11 @@ export class BaseProtocolSDK implements IBaseProtocolSDK {
       throw error;
     }
   }
+
+  private updatePeers(peers: Peer[]): void {
```

Comment:
nit: not to change everything else - we can introduce `private set peers` and do update inside